pkg/yurttunnel/server/server.go

```diff
@@ -20,12 +20,145 @@ import (
 	"context"
 	"crypto/tls"
 	"errors"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/google/uuid"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"k8s.io/klog"
+
+	anpserver "sigs.k8s.io/apiserver-network-proxy/pkg/server"
+	anpagent "sigs.k8s.io/apiserver-network-proxy/proto/agent"
 )
 
 // RunServer runs the yurttunnel-server
 func RunServer(ctx context.Context,
+	egressSelectorEnabled bool,
 	interceptorServerUDSFile,
-	serverMasterAddr, serverAgentAddr string,
+	serverMasterAddr,
+	serverMasterInsecureAddr,
+	serverAgentAddr string,
+	tlsCfg *tls.Config) error {
+	proxyServer := anpserver.NewProxyServer(uuid.New().String(), 1,
+		&anpserver.AgentTokenAuthenticationOptions{})
+
+	// 1. start the proxier
+	proxierErr := runProxier(&anpserver.Tunnel{Server: proxyServer},
+		egressSelectorEnabled,
+		interceptorServerUDSFile,
+		tlsCfg)
+	if proxierErr != nil {
+		return fmt.Errorf("fail to run the proxier: %s", proxierErr)
+	}
+
+	// 2. start the master server
+	masterServerErr := runMasterServer(
+		&RequestInterceptor{
+			UDSSockFile: interceptorServerUDSFile,
+			TLSConfig:   tlsCfg,
+		},
```

Comment:
would you add handler wrapper slice for master handler in order to modify request for user customization?

Reply:
Yes, I will add a request handler wrapper in the next pr.

Reply:
will add the handler wrapper slice in next pr.